test/jasmine/tests/cartesian_interact_test.js

```diff
@@ -2338,7 +2337,7 @@ describe('Event data:', function() {
             expect('marker.colorbar.tickvals' in pt).toBe(false, 'marker.colorbar.tickvals');
             expect('marker.colorbar.ticktext' in pt).toBe(false, 'marker.colorbar.ticktext');
         })
-        .catch(fail)
```

Comment:
Ah, good catch, this is a tricky one - for context, we used to call this function `fail` but changed it to `failTest` because `fail` is a global created by Jasmine to manually cause a test to fail. This mostly worked though it's not great to override a global. But the big problem came when we occasionally forgot to import our `fail_test` and ended up using the Jasmine global - which would just swallow the failure.

That said I notice there's a [`done.fail` and in fact in v3 smarter `done` behavior](https://jasmine.github.io/tutorials/async#failing-with-callbacks) that looks like it should make `failTest` obsolete - instead of `.catch(failTest).then(done)` we may be able to do just `.then(done, done.fail)` or perhaps even `.then(done, done)`

Definitely not something we need to do now, and it would take some experimentation to be sure this is going to fail correctly with at least as much information as `failTest` gives us.